### PR TITLE
DPRS-656: Updates to reg w/o id

### PR DESCRIPTION
**Pull request by paddynski-opencast**

### CODE REVIEW

File: test/uk/gov/hmrc/dprs/converters/registration/RegistrationConverterBaseSpec.scala [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.dprs.converters.registration
+
+import uk.gov.hmrc.dprs.connectors.registration.RegistrationConnector
+import uk.gov.hmrc.dprs.connectors.registration.RegistrationConnector.Request.Common
+import uk.gov.hmrc.dprs.services.BaseSpec
+
+abstract class RegistrationConverterBaseSpec extends BaseSpec {
```
Comment (erasmos-hmrc-new) [inline test/uk/gov/hmrc/dprs/converters/registration/RegistrationConverterBaseSpec.scala:23]: We don't need this.

File: test/uk/gov/hmrc/dprs/converters/registration/withoutId/RegistrationConverterWithoutIdForIndividualConverterSpec.scala [hunk 2]
```diff
@@ -54,12 +54,12 @@ class RegistrationWithoutIdForIndividualConverterSpec extends BaseSpec {
       val connectorRequest = converter.convert(serviceRequest)
 
       connectorRequest shouldBe ConnectorRequest(
-        common = RegistrationConnector.Request.Common(receiptDate = currentDateTime, regime = "MDR", acknowledgementReference = acknowledgementReference),
+        common = expectedCommon(),
```
Comment (erasmos-hmrc-new) [inline test/uk/gov/hmrc/dprs/converters/registration/withoutId/RegistrationConverterWithoutIdForIndividualConverterSpec.scala:57]: Inline object.

File: test/uk/gov/hmrc/dprs/converters/registration/withoutId/RegistrationConverterWithoutIdForOrganisationConverterSpec.scala [hunk 3]
```diff
@@ -46,7 +46,7 @@ class RegistrationWithoutIdForOrganisationConverterSpec extends BaseSpec {
       val connectorRequest = converter.convert(serviceRequest)
 
       connectorRequest shouldBe RegistrationWithoutIdForOrganisationConnector.Request(
-        common = RegistrationConnector.Request.Common(receiptDate = currentDateTime, regime = "MDR", acknowledgementReference = acknowledgementReference),
+        common = expectedCommon(),
```
Comment (erasmos-hmrc-new) [inline test/uk/gov/hmrc/dprs/converters/registration/withoutId/RegistrationConverterWithoutIdForOrganisationConverterSpec.scala:49]: Inline object.

File: test/uk/gov/hmrc/dprs/converters/registration/withId/RegistrationConverterWithIdForOrganisationConverterSpec.scala [hunk 4]
```diff
@@ -49,12 +49,7 @@ class RegistrationWithIdForOrganisationConverterSpec extends BaseSpec {
         val connectorRequest = converter.convert(serviceRequest)
 
         connectorRequest shouldBe RegistrationWithIdForOrganisationConnector.Request(
-          common = RegistrationConnector.Request.Common(
-            receiptDate = currentDateTime,
-            regime = "DPRS",
-            acknowledgementReference = acknowledgementReference,
-            requestParameters = Seq(RegistrationConnector.Request.Common.RequestParameter("REGIME", "DPRS"))
-          ),
+          common = expectedCommon(),
```
Comment (erasmos-hmrc-new) [inline test/uk/gov/hmrc/dprs/converters/registration/withId/RegistrationConverterWithIdForOrganisationConverterSpec.scala:52]: Inline.

File: test/uk/gov/hmrc/dprs/converters/registration/withoutId/RegistrationConverterWithoutIdForIndividualConverterSpec.scala (hunk 2 above)
Comment (erasmos-hmrc-new) [inline test/uk/gov/hmrc/dprs/converters/registration/withoutId/RegistrationConverterWithoutIdForIndividualConverterSpec.scala:57]: Inline.

File: test/uk/gov/hmrc/dprs/converters/registration/withId/RegistrationConverterWithIdForIndividualConverterSpec.scala [hunk 5]
```diff
@@ -40,12 +40,7 @@ class RegistrationWithIdForIndividualConverterSpec extends BaseSpec {
       val connectorRequest = converter.convert(serviceRequest)
 
       connectorRequest shouldBe ConnectorRequest(
-        common = RegistrationConnector.Request.Common(
-          receiptDate = currentDateTime,
-          regime = "DPRS",
-          acknowledgementReference = acknowledgementReference,
-          requestParameters = Seq(RegistrationConnector.Request.Common.RequestParameter("REGIME", "DPRS"))
-        ),
+        common = expectedCommon(),
```
Comment (erasmos-hmrc-new) [inline test/uk/gov/hmrc/dprs/converters/registration/withId/RegistrationConverterWithIdForIndividualConverterSpec.scala:43]: Inline.

File: app/uk/gov/hmrc/dprs/converters/registration/RegistrationConverter.scala [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    receiptDate = Instant.now(clock).toString,
+    regime = regime,
+    acknowledgementReference = acknowledgementReferenceGenerator.generate(),
+    requestParameters = Seq(RequestParameter("REGIME", regime))
+  )
+}
+
+object RegistrationConverter {
+
+  val regime = "DPRS"
```
Comment (erasmos-hmrc-new) [inline app/uk/gov/hmrc/dprs/converters/registration/RegistrationConverter.scala:39]: This only used in this class, so can go into the class, and made private.

```
final private val regime = "DPRS".
```

**Review by erasmos-hmrc-new — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/dprs/converters/registration/RegistrationConverter.scala (hunk 6 above)
Comment (paddynski-opencast) [reply app/uk/gov/hmrc/dprs/converters/registration/RegistrationConverter.scala]: Done.

File: test/uk/gov/hmrc/dprs/converters/registration/withoutId/RegistrationConverterWithoutIdForIndividualConverterSpec.scala (hunk 2 above)
Comment (paddynski-opencast) [reply test/uk/gov/hmrc/dprs/converters/registration/withoutId/RegistrationConverterWithoutIdForIndividualConverterSpec.scala]: Done.

File: test/uk/gov/hmrc/dprs/converters/registration/withId/RegistrationConverterWithIdForOrganisationConverterSpec.scala (hunk 4 above)
Comment (paddynski-opencast) [reply test/uk/gov/hmrc/dprs/converters/registration/withId/RegistrationConverterWithIdForOrganisationConverterSpec.scala]: Done.

File: test/uk/gov/hmrc/dprs/converters/registration/withoutId/RegistrationConverterWithoutIdForIndividualConverterSpec.scala (hunk 2 above)
Comment (paddynski-opencast) [reply test/uk/gov/hmrc/dprs/converters/registration/withoutId/RegistrationConverterWithoutIdForIndividualConverterSpec.scala]: Done.

File: test/uk/gov/hmrc/dprs/converters/registration/RegistrationConverterBaseSpec.scala (hunk 1 above)
Comment (paddynski-opencast) [reply test/uk/gov/hmrc/dprs/converters/registration/RegistrationConverterBaseSpec.scala]: Removed.

Comment (erasmos-hmrc-new) [reply test/uk/gov/hmrc/dprs/converters/registration/RegistrationConverterBaseSpec.scala]: Confirmed.

File: test/uk/gov/hmrc/dprs/converters/registration/withoutId/RegistrationConverterWithoutIdForIndividualConverterSpec.scala (hunk 2 above)
Comment (erasmos-hmrc-new) [reply test/uk/gov/hmrc/dprs/converters/registration/withoutId/RegistrationConverterWithoutIdForIndividualConverterSpec.scala]: Confirmed.

File: test/uk/gov/hmrc/dprs/converters/registration/withId/RegistrationConverterWithIdForOrganisationConverterSpec.scala (hunk 4 above)
Comment (erasmos-hmrc-new) [reply test/uk/gov/hmrc/dprs/converters/registration/withId/RegistrationConverterWithIdForOrganisationConverterSpec.scala]: Confirmed.

File: app/uk/gov/hmrc/dprs/converters/registration/RegistrationConverter.scala (hunk 6 above)
Comment (erasmos-hmrc-new) [reply app/uk/gov/hmrc/dprs/converters/registration/RegistrationConverter.scala]: Confirmed.

File: test/uk/gov/hmrc/dprs/converters/registration/withoutId/RegistrationConverterWithoutIdForOrganisationConverterSpec.scala (hunk 3 above)
Comment (erasmos-hmrc-new) [reply test/uk/gov/hmrc/dprs/converters/registration/withoutId/RegistrationConverterWithoutIdForOrganisationConverterSpec.scala]: Confirmed.

File: test/uk/gov/hmrc/dprs/converters/registration/withoutId/RegistrationConverterWithoutIdForIndividualConverterSpec.scala (hunk 2 above)
Comment (erasmos-hmrc-new) [reply test/uk/gov/hmrc/dprs/converters/registration/withoutId/RegistrationConverterWithoutIdForIndividualConverterSpec.scala]: Confirmed.

**Comment by erasmos-hmrc-new** [comment]
```
ACCESS_LOG_FILE appender (and usage) can be removed from ...
```

I think this isn't correct, as we can see a reference in the same file:

```
    <logger name="accesslog" level="INFO" additivity="false">
        <appender-ref ref="ACCESS_LOG_FILE" />
    </logger>
```

So let's leave it in.

File: etc/postman/RegistrationWithoutID.json [hunk 7]
```diff
@@ -319,6 +319,59 @@
 								}
 							]
 						},
+						{
+							"name": "Forbidden",
```
Comment (erasmos-hmrc-new) [inline etc/postman/RegistrationWithoutID.json:323]: Both of these 'forbidden' scenarios should be in the Failure / Downstream folders:

image

Comment (paddynski-opencast) [reply etc/postman/RegistrationWithoutID.json]: Moved both scenarios.

**Review by erasmos-hmrc-new — APPROVED** [review]

**Pull request merged**